#!/usr/bin/env python
PACKAGE = "pointcloud_utils"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("map_resolution", 	double_t, 0, "pointcloud grid resolution",    0.5, 0.01,   10)
gen.add("z_scale_max", 		double_t, 0, "bottom of the height scale",    -10, -100,   100)
gen.add("z_scale_min", 		double_t, 0, "top of the height scale",    	  10, -100,   100)
gen.add("map_width", 		int_t,    0, "number of cell columns in the grid", 1024, 2, 65536)
gen.add("map_height", 		int_t,    0, "number of cell rows in the grid", 1024, 2, 65536)
gen.add("x_centered", 		bool_t,   0, "Whether to place the origin at the x center or not", False)
gen.add("y_centered",  		bool_t,   0, "Whether to place the origin at the y center or not", True)

gen.add("x_min", 			double_t, 0, "Minimum bound in x to add points to the grid", -20 , -1000, 1000)
gen.add("x_max", 			double_t, 0, "Maximum bound in x to add points to the grid",  100, -1000, 1000)
gen.add("y_min", 			double_t, 0, "Minimum bound in y to add points to the grid", -30 , -1000, 1000)
gen.add("y_max", 			double_t, 0, "Maximum bound in y to add points to the grid",  30 , -1000, 1000)
gen.add("z_min", 			double_t, 0, "Minimum bound in z to add points to the grid", -5  , -1000, 1000)
gen.add("z_max", 			double_t, 0, "Maximum bound in z to add points to the grid",  5  , -1000, 1000)


exit(gen.generate(PACKAGE, "pointcloud_utils", "PointCloudUtils"))
